addons/project_timesheet_holidays/models/hr_holidays.py

```diff
@@ -32,11 +25,15 @@ def _compute_timesheet_generate(self):
         for leave_type in self:
             leave_type.timesheet_generate = leave_type.timesheet_task_id and leave_type.timesheet_project_id
 
+    @api.depends('company_id')
+    def _compute_timesheet_project_id(self):
+        for leave in self:
+            leave.timesheet_project_id = leave.company_id.internal_project_id or False
```

Comment:
```suggestion
            leave.timesheet_project_id = leave.company_id and leave.company_id.internal_project_id
```
I don't think the False is necessary here, and we have to handle the case where the leave type has no company